### PR TITLE
ensure footprints display when added before linking

### PR DESCRIPTION
**Pull request by gibsongreen**
### Description

This PR is to address when footprints are added via API prior to linking by WCS. Only the last footprint added was visible in the viewer once linked, even when all footprints are set to be visible. 

https://github.com/spacetelescope/jdaviz/assets/42986583/6160d9a8-16d7-46da-9012-67b976c2ce30

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/imviz/plugins/footprints/footprints.py
Comment (kecnry) [inline jdaviz/configs/imviz/plugins/footprints/footprints.py:192]: can we call the method internally so that the selected choice isn't affected (I'm assuming this works because it triggers the observe on `overlay_selected`)?  If that isn't possible, can we at least restore the user-selection after this for-loop?

Comment (gibsongreen) [reply jdaviz/configs/imviz/plugins/footprints/footprints.py]: It is triggering the observe. I am seeing that without changing the selected choice and calling the method internally, the only overlay that displays is the last one added based on `preset_args_changed`.

Comment (kecnry) [reply jdaviz/configs/imviz/plugins/footprints/footprints.py]: can we have the internal method take the name of the overlay and _default_ to overlay_selected so that we can trigger the logic without changing the selection?

Comment (kecnry) [reply jdaviz/configs/imviz/plugins/footprints/footprints.py]: see https://github.com/gibsongreen/jdaviz/pull/1 for a rough idea of how this could work (although I didn't have a test case, so haven't tested if it still fixes the bug - a regression test would be great if possible!)

File: jdaviz/configs/imviz/plugins/footprints/footprints.py
Comment (kecnry) [inline jdaviz/configs/imviz/plugins/footprints/footprints.py:193]: if there aren't any choices, then the for-loop shouldn't even enter.  It might also be a good idea to add a comment here so we don't think this is changing the selection (feel free to edit the text to whatever you think makes sense).

```suggestion
            for choice in self.overlay.choices:
                # trigger the update without actually changing the user-selection
                self._change_overlay(overlay_selected=choice)

```

**Review by kecnry — APPROVED** [review]
LGTM pending the one comment I left (which isn't crucial).  Thanks for tracking this down!

File: jdaviz/configs/imviz/plugins/footprints/footprints.py
Comment (gibsongreen) [reply jdaviz/configs/imviz/plugins/footprints/footprints.py:193]: When Imviz is instantiated in a notebook or test, the `default` overlay is temporarily in `self.overlay.choice`(and it is NoneType at this time). So to make sure that a TypeError exception isn't thrown I added this conditional.

I will add a comment now for some clarity on what this loop is doing.

Comment (kecnry) [reply jdaviz/configs/imviz/plugins/footprints/footprints.py]: so it enters the loop and then the choice disappears before `_change_overlay` is called?

Comment (gibsongreen) [reply jdaviz/configs/imviz/plugins/footprints/footprints.py]: The default does get passed into `_change_overlay`, it does go through creating a new entry, at the end of creating a new entry, since there is now 1 entry, the overlay tries to `center_on_viewer`. And when this happens the `viewer.get_center_skycoord` is what is actually NoneType (I mistakenly said it was the overlay). 

In the case that no overlay is added in the API before WCS linking, the default that gets created does still center in the viewer (so this does still trigger correctly if there is only 1 overlay).

File: jdaviz/configs/imviz/plugins/footprints/footprints.py
Comment (rosteen) [inline jdaviz/configs/imviz/plugins/footprints/footprints.py:189]: One nitpick, please write out "footprint" in the comments rather than using potentially opaque abbreviations, for future readers' sake.

Comment (gibsongreen) [reply jdaviz/configs/imviz/plugins/footprints/footprints.py]: I totally agree, change made!

**Review by rosteen — APPROVED** [review]
Approved but I'd prefer to see tests pass after #2809 gets merged before merging this.

Edit: eh, I guess as long as it's only the one know failure, go ahead and merge.

**Comment by gibsongreen** [comment]
> Approved but I'd prefer to see tests pass after #2809 gets merged before merging this.
> 
> Edit: eh, I guess as long as it's only the one know failure, go ahead and merge.

I just mad

> Approved but I'd prefer to see tests pass after #2809 gets merged before merging this.
> 
> Edit: eh, I guess as long as it's only the one know failure, go ahead and merge.

I just made a small change so this will go through the CI now and make sure they all pass!

**Pull request merged**
